core-bundle/src/Resources/contao/forms/Form.php

```diff
@@ -419,6 +419,10 @@ protected function processFormData($arrSubmitted, $arrLabels, $arrFields)
 			{
 				$email->attachFileFromString(StringUtil::decodeEntities('"' . implode('";"', $keys) . '"' . "\n" . '"' . implode('";"', $values) . '"'), 'form.csv', 'text/comma-separated-values');
 			}
+			elseif ($this->format == 'csv_excel')
+			{
+				$email->attachFileFromString(mb_convert_encoding("\u{FEFF}sep=;\n" . StringUtil::decodeEntities('"' . implode('";"', $keys) . '"' . "\n" . '"' . implode('";"', $values) . '"'), 'UTF-16LE', 'UTF-8'), 'form.csv', 'text/comma-separated-values');
```

Comment:
Shouldn't this be `\u{EFBBBF}` instead? `EFBBBF` would be the BOM for `UTF-8`, while `FEFF` is for `UTF-16` with big-endian byte order. And the contents of the file should be UTF-8, not UTF-16, according to the conversion.

Reply:
nvm, the conversion is _to_ UTF-16LE. However, as already mentioned, the BOM is for big-endian, while the conversion is to little-endian. So the BOM should be `FFFE`, shouldn't it?

But why do we convert to `UTF-16LE` in general? The pseudo-standard for cross-compatibility seems to be `\u{EFBBBF}sep=;` together with `UTF-8` data.

Reply:
There is no such thing as `\u{EFBBBF}` 🙃 You probably mean `\xEF\xBB\xBF` which is exactly the same as `\u{FEFF}` 🤓

Using an UTF-8 BOM together with `sep=` doesn’t work, Excel just ignores the BOM in this case and special characters don’t work anymore. `UTF-16LE` seems to be the only choice which works on all my tested devices (except Excel for macOS)

Reply:
Hm, I am confused, isn't `FE FF` the hexadecimal representation of the BOM in UTF-16 big-endian? And under UTF-8 it would be `EF BB BF`. And under UTF-16 little-endian it would be `FFFE`. See also https://en.wikipedia.org/wiki/Byte_order_mark#Byte_order_marks_by_encoding

Thus I would try using

```
mb_convert_encoding("\u{FFFE}sep=;\n" . StringUtil::decodeEntities('"' . implode('";"', $keys) . '"' . "\n" . '"' . implode('";"', $values) . '"'), 'UTF-16LE', 'UTF-8')
```

Reply:
Never mind my ramblings, I was confused about the `\u{FEFF}` which is simply the unicode representation of the BOM.